src/shared/utils.js

```diff
@@ -94,6 +96,16 @@ export const getTileClasses = ({
     classes.push(`${className}--hover`);
   }
 
+  if (!(isEqual(dateRange[0], valueRange[0]) && isEqual(dateRange[1], valueRange[1]))) {
```

Comment:
This function is called quite a lot. It would be better both for readability and for performance to cache some values. What do you think about that?

```js
const isRangeStart = isEqual(dateRange[0], valueRange[0]);
const isRangeEnd = isEqual(dateRange[1], valueRange[1]);

if (isRangeStart) {
  classes.push(`${className}--rangeStart`);
} else if (isRangeEnd) {
  classes.push(`${className}--rangeEnd`);
}

if (isRangeStart && isRangeEnd)
  classes.push(`${className}--rangeBothEnds`);
}
```